extensions/DrawNodeEx/DrawNodeEx.cpp

```diff
@@ -0,0 +1,1213 @@
+                {Vec2(vec2->x, vec2->y), fillColor, Tex2F(0.0, 0.0)},
+                {Vec2(vec3->x, vec3->y), fillColor, Tex2F(0.0, 0.0)},
+            };
+            *cursor++ = tmp;
+        }
+    }
+    else
+    {
+        for (int i = 0; i < count - 2; i++)
+        {
+            V2F_C4B_T2F_Triangle tmp = {
+                {_vertices[0], fillColor, v2ToTex2F(Vec2::ZERO)},
+                {_vertices[i + 1], fillColor, v2ToTex2F(Vec2::ZERO)},
+                {_vertices[i + 2], fillColor, v2ToTex2F(Vec2::ZERO)},
+            };
+
+            *cursor++ = tmp;
+        }
+    }
+    if (outline)
+    {
+        struct ExtrudeVerts
+        {
+            Vec2 offset, n;
+        };
+        struct ExtrudeVerts* extrude = (struct ExtrudeVerts*)malloc(sizeof(struct ExtrudeVerts) * count);
+
+        for (int i = 0; i < count; i++)
+        {
+            Vec2 v0 = _vertices[(i - 1 + count) % count];
+            Vec2 v1 = _vertices[i];
+            Vec2 v2 = _vertices[(i + 1) % count];
+
+            Vec2 n1 = ((v1 - v0).getPerp()).getNormalized();
+            Vec2 n2 = ((v2 - v1).getPerp()).getNormalized();
+
+            Vec2 offset = (n1 + n2) * (1.0f / (Vec2::dot(n1, n2) + 1.0f));
+            extrude[i]  = {offset, n2};
+        }
+
+        for (int i = 0; i < count; i++)
+        {
+            int j   = (i + 1) % count;
+            Vec2 v0 = _vertices[i];
+            Vec2 v1 = _vertices[j];
+
+            Vec2 n0 = extrude[i].n;
+
+            Vec2 offset0 = extrude[i].offset;
+            Vec2 offset1 = extrude[j].offset;
+
+            Vec2 inner0 = v0 - offset0 * borderWidth;
+            Vec2 inner1 = v1 - offset1 * borderWidth;
+            Vec2 outer0 = v0 + offset0 * borderWidth;
+            Vec2 outer1 = v1 + offset1 * borderWidth;
+            borderColor = borderColo;
+            if (i >= count - 1 && !closedPolygon)  // /-2  ??
+            {
+                borderColor = Color4B::AX_TRANSPARENT;
+            }
+
+            V2F_C4B_T2F_Triangle tmp1 = {{inner0, borderColor, v2ToTex2F(-n0)},
+                                         {inner1, borderColor, v2ToTex2F(-n0)},
+                                         {outer1, borderColor, v2ToTex2F(n0)}};
+            *cursor++                 = tmp1;
+
+            V2F_C4B_T2F_Triangle tmp2 = {{inner0, borderColor, v2ToTex2F(-n0)},
+                                         {outer0, borderColor, v2ToTex2F(n0)},
+                                         {outer1, borderColor, v2ToTex2F(n0)}};
+            *cursor++                 = tmp2;
+        }
+
+        free(extrude);
+    }
+
+    _customCommandTriangle.updateVertexBuffer(triangles, _bufferCountTriangle * sizeof(V2F_C4B_T2F),
+                                              vertex_count * sizeof(V2F_C4B_T2F));
+    _bufferCountTriangle += vertex_count;
+    _customCommandTriangle.setVertexDrawInfo(0, _bufferCountTriangle);
+    _dirtyTriangle = true;
+}
+
+inline void DrawNodeEx::_drawPoly(const Vec2* poli,
```

Comment:
```suggestion
void DrawNodeEx::_drawPoly(const Vec2* poli,
```

Reply:
@halx99 
`inline` I thought makes a better performance, isnt it?

Reply:
No， The code large, not recommnd add inline keyword, and the compiler actually ignore the inline mark when a function to large or logic too complex